sweeper/sweeper.go

```diff
@@ -0,0 +1,191 @@
+package sweeper
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/skynetlabs/pinner/database"
+	"github.com/skynetlabs/pinner/skyd"
+	"gitlab.com/NebulousLabs/errors"
+	"gitlab.com/SkynetLabs/skyd/build"
+	"gitlab.com/SkynetLabs/skyd/skymodules"
+)
+
+type (
+	// Schedule defines how often, if at all, we sweep this server automatically.
+	Schedule struct {
+		Period   time.Duration
+		cancelCh chan struct{}
+	}
+	// Status represents the status of a sweep.
+	Status struct {
+		InProgress bool
+		Error      error
+		StartTime  time.Time
+		EndTime    time.Time
+	}
+	// Sweeper takes care of sweeping the files pinned by the local skyd server
+	// and marks them as pinned by the local server.
+	Sweeper struct {
+		staticDB         *database.DB
+		staticSkydClient skyd.Client
+		staticServerName string
+
+		schedule   *Schedule
+		scheduleMu sync.Mutex
```

Comment:
If you have a mutex guarding a nested struct, you can make it `static` and push the mutex into the struct, which makes more sense because it's only guarding those fields anyway. In `skyd` we have the `call` and `managed` conventions to go alongside it so you and up with 

```go
sweeper.staticSchedule.callUpdatePeriod(newPeriod)
```
instead of 
```go
sweeper.scheduleMu.Lock()
sweeper.schedule.period = newPeriod
sweeper.scheduleMu.Unlock()
```

which is nice because you push the locking behaviour and the management into the sub struct so you don't have to care about it in your parent struct, you only have to pay attention that you're not holding two locks at the same time.